README.md

```diff
@@ -30,27 +51,249 @@ npm install --save ipfsd-ctl
+
+## IPFS executable types
+
+`ipfsd-ctl` allows spawning different types of executables, such as:
+
+> `go`
+
+Invoking `df.spawn({type: 'go', exec: '<path to go executable or empty to allow ipfsd to find one automatically>'})` will spawn a `go-ipfs` node.
+
+> `js`
+
+Invoking `df.spawn({type: 'js', exec: '<path to js executable or empty to allow ipfsd to find one automatically>'})` will spawn a `js-ipfs` node.
+
+> `proc`
+
+Invoking `df.spawn({type: 'proc', exec: require('ipfs')})` will spawn an `in-process-ipfs` node using the provided code reference that implements the core IPFS API. Note that, `exec` is required if `type: 'proc'` is used.
+
+### IPFS executables
+
+`ipfsd-ctl` no longer installs go-ipfs nor js-ipfs dependencies, instead it expects them to be provided by the parent project. In order to be able to use both go and js daemons, please make sure that your project includes these two npm packages as dependencies.
+
+- `ipfs` - the js-ipfs implementation
+- `go-ipfs-dep` - the packaged go-ipfs implementation
+
+## API
+
+### Daemon Factory
+
+#### Create a `DaemonFactory`
+
+> `DaemonFactory.create([options])` create a factory that will expose the `df.spawn` method
+
+- `options` - an optional object with the following properties
+  - `remote` bool - indicates if the factory should spawn local or remote nodes. By default, local nodes are spawned in Node.js and remote nodes are spawned in Browser environments.
+  - `port` number - the port number to use for the remote factory. It should match the port on which `DaemonFactory.server` was started. Defaults to 9999.
+
+> `DaemonFactory.createServer` create an instance of the bundled HTTP server used by the remote controller.
+
+- exposes `start` and `stop` methods to start and stop the http server.
+
+#### Spawn a new daemon with `df.spawn`
+
+> Spawn either a js-ipfs or go-ipfs node
+
+`spawn([options], callback)`
+
+- `options` - is an optional object the following properties
+  - `type` string (default 'go') - indicates which type of node to spawn
+    - current valid values are `js`, `go` and `proc`
+  - `init` bool (default true) - should the node be initialized
+  - `start` bool (default true) - should the node be started
+  - `repoPath` string - the repository path to use for this node, ignored if node is disposable
+  - `disposable` bool (default false) - a new repo is created and initialized for each invocation, as well as cleaned up automatically once the process exits
+  - `args` - array of cmd line arguments to be passed to ipfs daemon
+  - `config` - ipfs configuration options
+  - `exec` - path to the desired IPFS executable to spawn, otherwise `ipfsd-ctl` will try to locate the correct one based on the `type`. In the case of `proc` type, exec is required and expects an IPFS coderef
```

Comment:
Should be part of the DaemonFactory options.

Reply:
moved relevant options to `DaemonFactory.create()`